figures/README.md

```diff
@@ -41,3 +42,121 @@ However, we list information about the resources, intermediate steps, and [PBTA
+|`binary_col_palette.tsv` |<br>binary_1:![2166ac](https://placehold.it/150x40/2166ac/FFFFFF?text=2166ac) <br>binary_2:![b2182b](https://placehold.it/150x40/b2182b/FFFFFF?text=b2182b) <br>na_color:![f1f1f1](https://placehold.it/150x40/f1f1f1/000000?text=f1f1f1)|A vector of two hex codes|For binary variables e.g. presence/absence or Amp/Del as statuses|
+
+
+## Color coding examples in R
+
+#### Example 1) Color coding by `short_histology`.
+
+**Step 1)** Read in color palette and format as a named list
+```
+histology_col_palette <- readr::read_tsv(
+  file.path("figures", "palettes", "histology_color_palette.tsv")
+  ) %>%
+  # We'll use deframe so we can use it as a recoding list
+  tibble::deframe()
+```
+
+**Step 2)** For any data.frame with a `short_histology` column, recode NAs as "none".
+```
+metadata <- readr::read_tsv(file.path("data", "pbta-histologies.tsv") %>%
+  # Easier to deal with NA short histologies if they are labeled something different
+  dplyr::mutate(short_histology = as.character(tidyr::replace_na(short_histology, "none")))
+```
+
+**Step 3)** Use dplyr::recode on `short_histology` column to make a new color column.
+```
+metadata <- metadata %>%
+  # Tack on the sample color using the short_histology column and a recode
+  dplyr::mutate(sample_color = dplyr::recode(short_histology,
+                                             !!!histology_col_palette))
+```
+
+**Step 4)** Make your plot and use the `sample_color` column.  
+Using the `ggplot2::scale_fill_identity()` or `ggplot2::scale_color_identity()`
+allows you to supply an exact `hex_code` column to `ggplot2` with a `fill` or
+`color` argument respectively.
+For base R plots, you should be able to supply the `sample_color` column as your
+`col` argument.
+```
+metadata %>%
+  dplyr::group_by(short_histology, sample_color) %>%
+  dplyr::summarize(count = dplyr::n()) %>%
+  ggplot2::ggplot(ggplot2::aes(x = short_histology, y = count, fill = sample_color)) +
+  ggplot2::geom_bar(stat = "identity") +
+  ggplot2::scale_fill_identity()
+```
+
+#### Example 2) Color coding by numeric data
+
+**Step 1)** Import the palette.
+
+You may want to remove the `na_color` at the end of the list depending on what
+your data look like.
+```
+gradient_col_palette <- readr::read_tsv(
+  file.path(figures_dir, "palettes", "gradient_color_palette.tsv")
+  )
+```
+
+If we need the `NA` color separated, like for use with `ComplexHeatmap` which has
+a separate argument for the color for `NA` values.
+
+```
+na_color <- gradient_col_palette %>%
+  dplyr::filter(color_names != "na_color")
+
+gradient_col_palette <- gradient_col_palette %>%
+  dplyr::filter(color_names != "na_color")
+```
+
+**Step 2)** Make a color function.  
+
+The numbers supplied here will be highly dependent on what your data's distribution looks like.
```

Comment:
```suggestion
The numbers supplied here will be highly dependent on the distribution of the data you are plotting.
```

Reply:
Supplied here or generated by `seq`? Do you mean for folks to change the `from` or `to` or are you talking about what ends up stored in `gradient_col_val`?

Reply:
Yes. My point is `seq` gives you a regular interval based on this example, but perhaps a regular interval for color coding your data isn't going to be what you want. e.g. if your data is on an exponential scale, a regular interval -based palette might not represent your data well on the plot.